### PR TITLE
fix(deps): upgrade knex

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes NAN-594

- Upgrade and pin knex

### CODE REVIEW

File: packages/jobs/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
         "directory": "packages/jobs"
     },
     "dependencies": {
+        "@nangohq/nango-jobs": "file:../jobs",
```
Comment (TBonnin) [inline packages/jobs/package.json]: depends on itself? 😛

File: packages/jobs/package.json [hunk 2]
```diff
@@ -29,7 +30,7 @@
         "dotenv": "^16.0.3",
         "fs-extra": "^11.1.1",
         "js-yaml": "^4.1.0",
-        "knex": "^2.4.2",
+        "knex": "3.1.0",
```
Comment (TBonnin) [inline packages/jobs/package.json:32]: it is a major upgrade. Are we fully confident upgrading doesn't break anythin?

File: packages/jobs/package.json (hunk 1 above)
Comment (bodinsamuel) [reply packages/jobs/package.json]: ah jesus it was a test, thanks

File: packages/jobs/package.json (hunk 2 above)
Comment (bodinsamuel) [reply packages/jobs/package.json:32]: the main breaking change is the drop of Node16 and some small types fixes 
https://github.com/knex/knex/releases/tag/3.0.0

But I would love to test on staging before merging

**Comment by bodinsamuel** [comment]
Looks good on staging 👌🏻

**Review by TBonnin — APPROVED** [review]

**Pull request merged**